### PR TITLE
HDDS-11112. Verify javadoc creation in CI

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Javadoc is generated only with the `dist` profile.  This is used in CI build, but Javadoc is explicitly skipped.  Thus, javadoc generation is currently exercised only when making release candidates.

This change enables Javadoc generation in the `compile` check.  Build is successful despite Javadoc problems, e.g.:

```
Warning:  .../GenericTestUtils.java:339: warning - invalid usage of tag >
```

We can enable `failOnWarning` after fixing these warnings.  The current change would help catch problems with usage of `maven-javadoc-plugin` itself (e.g. wrong version or config).

https://issues.apache.org/jira/browse/HDDS-11112

## How was this patch tested?

Verified `maven-javadoc-plugin` is executed in `compile` check:
https://github.com/adoroszlai/ozone/actions/runs/9828984546/job/27133798734#step:8:3488

### CODE REVIEW

File: hadoop-ozone/dev-support/checks/build.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ cd "$DIR/../../.." || exit 1
 
 : ${OZONE_WITH_COVERAGE:="false"}
 
-MAVEN_OPTIONS='-V -B -Dmaven.javadoc.skip=true -DskipTests -DskipDocs --no-transfer-progress'
+MAVEN_OPTIONS='-V -B -DskipTests -DskipDocs --no-transfer-progress'
```
Comment (sadanand48) [inline hadoop-ozone/dev-support/checks/build.sh:21]: build.sh is invoked in other places like intermittent-test-check.yml, repeat-acceptance.yml, build-ratis.yml . We would need to update skipjavadoc param in these places to retain same behaviour or undo this change

Comment (adoroszlai) [reply hadoop-ozone/dev-support/checks/build.sh:21]: Thanks for the reminder, updated.

Note that `build-ratis` invokes `build.sh` which comes with Ratis, not this one.

**Review by sadanand48 — APPROVED** [review]
Thanks @adoroszlai for the patch, LGTM.

**Comment by adoroszlai** [comment]
Thanks @sadanand48 for the review.
